### PR TITLE
Update setup.sh and cleanup.sh

### PR DESCRIPTION
**Pull request by marko7460**
This PR fixes setup.sh and cleanup.sh to:
1. Handle shared-vpc use case
2. Update roles on SA account
3. Enable necessary services on project in order for terraform to run without problems

### CODE REVIEW

**Comment by adrienthebo** [comment]
We're about to release v1.2.0 of this module to match to the forseti-security v2.12.0 release; in the interest in minimizing risk to this release I'm going to defer this until v1.2.1 or v1.3.0. My quick review of this is positive, will take it for a more thorough spin after v1.2.0 is out. Thanks!

File: helpers/cleanup.sh [hunk 1]
```diff
@@ -13,8 +13,60 @@
...
+            show_help >&2
+            exit 1
+            ;;
+    esac
+done
+shift "$((OPTIND-1))"   # Discard the options and sentinel --
+
+if [[ $project_id == "" ]];
+then
+    echo "ERROR -p option is mandatory"
```
Comment (aaron-lane) [inline helpers/cleanup.sh]: An option being mandatory is a contradiction. Mandatory arguments should be positional.

File: helpers/cleanup.sh [hunk 2]
```diff
@@ -13,8 +13,60 @@
...
+    exit 1
+fi
+
+if [[ $service_account_id == "" ]];
+then
+    echo "ERROR -s option is mandatory"
```
Comment (aaron-lane) [inline helpers/cleanup.sh]: Ditto about mandatory options.

File: helpers/setup.sh [hunk 3]
```diff
@@ -13,18 +13,63 @@
...
+            show_help >&2
+            exit 1
+            ;;
+    esac
+done
+shift "$((OPTIND-1))"   # Discard the options and sentinel --
+
+if [[ $project_id == "" ]];
+then
+    echo "ERROR -p option is mandatory"
```
Comment (aaron-lane) [inline helpers/setup.sh]: Ditto about mandatory options.

File: helpers/setup.sh [hunk 4]
```diff
@@ -35,6 +80,21 @@ SERVICE_ACCOUNT_ID="${SERVICE_ACCOUNT_NAME}@${PROJECT_ID}.iam.gserviceaccount.co
 STAGING_DIR="${PWD}"
 KEY_FILE="${STAGING_DIR}/credentials.json"
 
+if [[ ${host_project} != "" ]];
```
Comment (aaron-lane) [inline helpers/setup.sh]: Logic which is duplicated between `setup.sh` and `cleanup.sh` should be extracted to a shared library file.

Comment (adrienthebo) [reply helpers/setup.sh]: I'm normally a big proponent for DRY, but for things like simple shell scripts I tend to relax that a bit. Using a shared library reduces the amount of duplicated code but incurs the complexity cost of loading that shared library, which can be pretty obnoxious depending on the bash version and whatnot. We only have two copies of this logic, so I think that we should push back on this refactoring. If we need in the future let's add it, but let's wait for the demand to present itself.

File: helpers/cleanup.sh (hunk 1 above)
Comment (adrienthebo) [reply helpers/cleanup.sh]: Argument parsing styles are pretty subjective; I've seen (and written) a number of CLI tools that have used both positional arguments and required flags. Requiring flags for values is slightly more verbose but increases readability, which seems like a decent tradeoff.

In addition argument parsing in bash is generally less than pleasant, and mixing positional arguments with flags is generally a nightmare. For the sake of simplicity and readability, I actually prefer the way that we have this now. Is there a cost to this approach that I'm not seeing?

File: helpers/setup.sh (hunk 4 above)
Comment (adrienthebo) [reply helpers/setup.sh]: After additional review I've seen small issues that affect option parsing in both setup and cleanup; this makes me shift a bit closer towards having a shared file containing common definitions.

**Pull request merged**